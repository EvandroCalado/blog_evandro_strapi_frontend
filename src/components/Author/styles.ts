import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    display: flex;
    align-items: center;
    margin: ${theme.spacings.xl} auto;

    @media screen and ${theme.media.lteOrEqSmall} {
      flex-direction: column;
      width: 30rem;
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 22rem;
    height: 17rem;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;
    margin-right: ${theme.spacings.md};

    & img {
      object-fit: cover;
    }

    @media screen and ${theme.media.lteOrEqSmall} {
      width: 15rem;
      margin-bottom: ${theme.spacings.lg};
    }
`}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;

    @media screen and ${theme.media.lteOrEqSmall} {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
    }
`}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.sm} 0;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 3;
`}
`;
