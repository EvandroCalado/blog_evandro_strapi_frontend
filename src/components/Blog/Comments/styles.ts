import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: 0 auto;

    iframe {
      color: ${theme.colors.text};
    }
`}
`;
