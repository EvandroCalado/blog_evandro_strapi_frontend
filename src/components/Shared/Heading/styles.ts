import styled, { css } from 'styled-components';
import { HeadingProps } from '.';

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ uppercase = false }) => css`
    ${titleCase(uppercase)}
`}
`;
