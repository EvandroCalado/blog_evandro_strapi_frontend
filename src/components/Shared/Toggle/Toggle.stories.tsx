import { Meta } from '@storybook/react';
import Toggle from './Toggle';

export default {
  title: 'Components/Shared/Toggle',
  component: Toggle,
} as Meta;

export const Default = () => <Toggle />;
