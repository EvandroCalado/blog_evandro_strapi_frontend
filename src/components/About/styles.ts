import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    margin: 0 auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    gap: ${theme.spacings.xl};

    ${Title} {
      font-size: ${theme.font.sizes['4xl']};
    }
`}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.md};

    &::first-letter {
      font-size: ${theme.font.sizes.xl};
      font-weight: bold;
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`

    & > img {
      width: 35rem;
      height: 35rem;
      overflow: hidden;
      border-radius: ${theme.spacings.md};
      object-fit: cover;
    }
`}
`;
