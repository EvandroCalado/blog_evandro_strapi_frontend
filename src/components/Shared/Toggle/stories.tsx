import { Meta } from '@storybook/react';
import Toggle from '.';

export default {
  title: 'Components/Shared/Toggle',
  component: Toggle,
} as Meta;

export const Default = () => <Toggle />;
