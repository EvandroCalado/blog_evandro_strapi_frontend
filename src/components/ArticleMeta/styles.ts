import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['7xl']};
    margin: auto;
    padding: 0 ${theme.spacings.md};
    display: flex;
    align-items: center;
    flex-direction: column;
    gap: ${theme.spacings.xl};

    & > span > a {
      font-weight: bold;
      transition: ${theme.transitions.faster};
      margin-left: ${theme.spacings.sm};

      &:hover {
        text-decoration: underline;
      }
    }

    ${Title} {
      width: 100%;
      max-width: ${theme.frameSizes['4xl']};
      text-align: center;
    }
`}
`;

export const Image = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 40rem;
    object-fit: cover;
    border-radius: ${theme.spacings.sm};
    overflow: hidden;
  `}
`;

export const ArticleContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.frameSizes['4xl']};
    display: flex;
    align-items: center;
    justify-content: space-between;

    @media screen and ${theme.media.lteOrEqSmall} {
      flex-direction: column;
    }
`}
`;

export const DateContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: end;
    justify-content: center;
    gap: ${theme.spacings.lg};
`}
`;
