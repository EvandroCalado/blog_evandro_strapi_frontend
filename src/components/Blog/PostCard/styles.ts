import styled, { css } from 'styled-components';
import { Title } from '../../Shared/Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    flex-direction: column;
    padding: ${theme.spacings.lg} ${theme.spacings.md};

    & > a {
      width: 100%;
      height: 20rem;
      margin-bottom: ${theme.spacings.md};
    }
`}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 100%;
    margin-bottom: ${theme.spacings.md};

    & img {
      object-fit: cover;
      border-radius: ${theme.spacings.sm};
    }
`}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.md};

    & > a {
      transition: ${theme.transitions.faster};

      &:hover {
        text-decoration: underline;
        filter: invert(.2);
      }
    }

    ${Title} {
      overflow: hidden;
      display: -webkit-box;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 2;
    }
`}
`;

export const Content = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.xs} 0;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 4;
`}
`;

export const DateContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
`;

export const Date = styled.span`
`;
