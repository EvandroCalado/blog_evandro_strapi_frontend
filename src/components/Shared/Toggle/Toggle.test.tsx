import '@testing-library/jest-dom';
import { fireEvent } from '@testing-library/react';
import { BlogThemeContext } from '../../../context/BlogThemeContext';
import { renderTheme } from '../../../styles/render-theme';
import Toggle from './Toggle';

describe('<Toggle />', () => {
  it('should render with default values', () => {
    const { container } = renderTheme(<Toggle />);

    expect(localStorage.getItem('theme')).toBe('light');
    const span = container.querySelector('span');

    fireEvent.click(span as Element);

    localStorage.setItem('theme', 'dark');
    expect(localStorage.getItem('theme')).toBe('dark');

    fireEvent.click(span as Element);

    localStorage.setItem('theme', 'light');
    expect(localStorage.getItem('theme')).toBe('light');
  });

  it("should set the theme based on the user's dark mode preference", () => {
    const setIsDarkMode = jest.fn();
    const isDarkMode = true;

    renderTheme(
      <BlogThemeContext.Provider value={{ isDarkMode, setIsDarkMode }}>
        <Toggle />
      </BlogThemeContext.Provider>,
    );

    expect(localStorage.getItem('theme')).toBe('dark');
  });

  it('should render a snapshot', () => {
    const { container } = renderTheme(<Toggle />);

    expect(container).toMatchSnapshot();
  });
});
