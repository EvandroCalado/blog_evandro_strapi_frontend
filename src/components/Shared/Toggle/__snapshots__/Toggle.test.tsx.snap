// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> should render a snapshot 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  width: 60px;
  height: 34px;
}

.c0 svg {
  position: absolute;
  inset: 1.6px;
  width: 3rem;
}

.c0 svg:last-of-type {
  transform: translateX(26.5px);
}

.c2 {
  opacity: 0;
  width: 0;
  height: 0;
}

.c3 {
  cursor: pointer;
  position: absolute;
  z-index: 5;
  inset: 0;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 34px;
  border: 1px solid;
}

.c3:before {
  position: absolute;
  content: '';
  height: 25.5px;
  width: 25.5px;
  left: 4px;
  bottom: 4px;
  background-color: #020415;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  border-radius: 50%;
}

.c1:checked+.c3:before {
  -webkit-transform: translateX(25px);
  -ms-transform: translateX(25px);
  transform: translateX(25px);
}

<div>
  <label
    class="c0"
  >
    <input
      class="c1 c2"
      type="checkbox"
    />
    <span
      class="c3"
    />
    <svg
      class="icon icon-tabler icon-tabler-sun"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="{1}"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <path
        d="M12 12m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0"
      />
      <path
        d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7"
      />
    </svg>
    <svg
      class="icon icon-tabler icon-tabler-moon"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="{1}"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <path
        d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
      />
    </svg>
  </label>
</div>
`;
